### PR TITLE
Update stack

### PR DESCRIPTION
**Pull request by micoli**

### CODE REVIEW

**Comment by micoli** [comment]
Hello, i've just opened that MR to update the stack of sylar, up to now it's blocked by morgonus/docker-api-php-client, which is not compatible with symfony 6.x . So I suggest to internalize that lib (may be in the future it can have it's dedicated repository. Without that blocker, the stack can be upgraded to php 8.2/symfony 6.2


What do you think also to remove the vagrant local development configuration. It's more a blocker than a real benefit to promove that project ?

**Comment by BenMorel** [comment]
Hey @micoli!

> up to now it's blocked by morgonus/docker-api-php-client, which is not compatible with symfony 6.x . So I suggest to internalize that lib (may be in the future it can have it's dedicated repository.

As an alternative, have you thought about opening a PR with `docker-api-php-client` to make it compatible with Symfony 6? Or considered using [another library](https://github.com/theodorosploumis/awesome-docker-php)? (haven't checked those personally, but they might be worth a look)

> What do you think also to remove the vagrant local development configuration. It's more a blocker than a real benefit to promote that project ?

What's the alternative?

**Comment by micoli** [comment]
> Hey @micoli!
> 
> > up to now it's blocked by morgonus/docker-api-php-client, which is not compatible with symfony 6.x . So I suggest to internalize that lib (may be in the future it can have it's dedicated repository.
> 
> As an alternative, have you thought about opening a PR with `docker-api-php-client` to make it compatible with Symfony 6? Or considered using ~[another library](https://github.com/theodorosploumis/awesome-docker-php)? (haven't checked those personally, but they might be worth a look)~
> 
> **edit**: this list and the associated projects are severely outdated, but this fork of `docker-php` looks worth investigating:
> 
> https://github.com/beluga-php/docker-php

I tried to integrate beluga-php, but it's not yet production ready. To help sylar, i've made a fork of it,(https://github.com/micoli/docker-php-api-client) and that is version that is now integrated in sylar. Sorry, the first comment it's quite old now. 

> 
> > What do you think also to remove the vagrant local development configuration. It's more a blocker than a real benefit to promote that project ?
> 
> What's the alternative?

I've updated the README.md (section `Development Virtual Machine setup`) to explain how to create a development environment on sylar, IMO, it shoudl be more productive to understand how the environment is created rather than having a magic recipe, without understanding the hows and whys.

File: data/sylar.yaml [hunk 1]
```diff
@@ -45,6 +45,34 @@ services:
...
+    lifeCycleHooks:
+      postStartWaiters:
+        - type: logMatch
+          expression: '!database system is ready to accept connections!'
+          timeout: 30
+    mounts:
+      - source: '=containerParameter.getReplicatedFilesystem()'
+        preCreateSource: false
+        target: /var/lib/postgresql/data
+#      - source: '=(configurationRoot)~"/"~((containerParameter.getIndex() === 0)?"mysql-master/etc/mysql/my.cnf":"mysql-replicate/etc/mysql/my.cnf")'
```
Comment (sereinity) [inline data/sylar.yaml]: to rm

File: .eslintrc.cjs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+//    "jsx-a11y/control-has-associated-label": "off",
+//    "jsx-a11y/no-static-element-interactions": "off",
+//    "no-console": "off",
+//    "no-param-reassign": "off",
+//    "no-plusplus": "off",
+//    "react-hooks/exhaustive-deps": "off",
+//    "react/forbid-prop-types": "off",
+//    "react/jsx-filename-extension": "off",
+//    "react/jsx-props-no-spreading": "off",
+//    "react/require-default-props": "off"
```
Comment (sereinity) [inline .eslintrc.cjs]: to rm?

File: assets/graphQL/GraphQL.ts [hunk 3]
```diff
@@ -1,42 +1,31 @@
...
+//       'Content-Type': 'application/json',
+//       ...headers
+//     },
+//     body: JSON.stringify({ query: mutationPayload })
+//   });
+// }
+//
+// const data = new FormData();
+// data.append('query', mutationPayload);
+// files.forEach((file) => data.append('files[]', file));
```
Comment (sereinity) [inline assets/graphQL/GraphQL.ts]: to rm?

File: .docker/mercure/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    php -r "if (hash_file('sha384', 'composer-setup.php') === 'e21205b207c3ff031906575712edab6f13eb0b361f2085f1f1237b7126d785e826a450292b6cfd1d64d92e6563bbde02') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;";\
+    php composer-setup.php;\
+    php -r "unlink('composer-setup.php');";\
+    mv composer.phar /usr/local/bin/composer
+
+RUN mkdir /app; \
+    cd /app; \
+    git clone https://github.com/bpolaszek/freddie; \
+    cd freddie; \
+    git checkout 6fc70c78fea27c7feda1aeb22486de71a52e0fa2; \
```
[the remark this answers is not in the document]
Comment (micoli) [reply .docker/mercure/Dockerfile:34]: indeed it's true for a branch, but not for a commit. At least i didn't find out how to do so

File: .docker/mercure/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    composer install --no-scripts
```
[the remark this answers is not in the document]
Comment (micoli) [reply .docker/mercure/Dockerfile:35]: done

File: .docker/mercure/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+cd /app/freddie
+#which make || apt-get install -y make
+composer install
```
[the remark this answers is not in the document]
Comment (micoli) [reply .docker/mercure/entrypoint.sh:5]: removed, no necessary

File: docker-compose.yaml [hunk 7]
```diff
@@ -4,6 +4,7 @@ volumes:
   builder-build: {}
   node-modules-builder: {}
   vendor-sylar: {}
+  vendor-mercure: {}
```
[the remark this answers is not in the document]
Comment (micoli) [reply docker-compose.yaml:9]: volume removed

File: docker-compose.yaml [hunk 8]
```diff
@@ -112,17 +113,23 @@ services:
 
   mercure:
     container_name: sylar-mercure
+    build: .docker/mercure
+    entrypoint: ["/entrypoint.sh"]
     restart: unless-stopped
-    image: dunglas/mercure:v0.10.4
+    working_dir: /app/freddie
```
[the remark this answers is not in the document]
Comment (micoli) [reply docker-compose.yaml:140]: done

File: docker-compose.yaml [hunk 9]
```diff
@@ -112,17 +113,23 @@ services:
...
     environment:
-      - ALLOW_ANONYMOUS=1
-      - CORS_ALLOWED_ORIGINS=*
-      - MERCURE_EXTRA_DIRECTIVES=cors_allowed_origins *
-      - PUBLISHER_JWT_KEY=GHJghjdjYYu1121kjkdstyydsghhdshzhgeh
-      - PUBLISH_ALLOWED_ORIGINS=*
-      - SUBSCRIBER_JWT_KEY=mercure_subscriber
-      - SERVER_NAME=":80"
-      - USE_FORWARDED_HEADERS=1
+      - APP_ENV=dev
```
[the remark this answers is not in the document]
Comment (micoli) [reply docker-compose.yaml:142]: done

File: data/sylar.yaml [hunk 10]
```diff
@@ -41,6 +41,7 @@ services:
           command: ['ls','/']
     mounts:
       - source: '=containerParameter.getReplicatedFilesystem()~"/data"'
+        preCreateSource: true
```
[the remark this answers is not in the document]
Comment (micoli) [reply data/sylar.yaml]: done

File: src/Core/ServiceCloner/Reservation/ReservationRepository.php [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+    //                ),
+    //                null,
+    //                null,
+    //                new PhpDocExtractor(),
+    //            ),
+    //        ], [
+    //            new JsonEncoder(),
+    //        ]);
+    //    }
+}
```
[the remark this answers is not in the document]
Comment (micoli) [reply src/Core/ServiceCloner/Reservation/ReservationRepository.php]: done

File: package.json [hunk 12]
```diff
@@ -4,18 +4,12 @@
   "licence": "MIT",
   "version": "1.0.0",
   "private": false,
+  "type": "module",
```
[the remark this answers is not in the document]
Comment (micoli) [reply package.json:7]: done

File: .eslintrc.cjs (hunk 2 above)
Comment (micoli) [reply .eslintrc.cjs]: done

File: data/sylar.yaml (hunk 1 above)
Comment (micoli) [reply data/sylar.yaml]: done

File: data/sylar.yaml [hunk 13]
```diff
@@ -14,8 +14,6 @@ services:
       value: sylar_user
     - name: MYSQL_PASSWORD
       value: sylar_password
-#    - name: MYSQL_DATABASE
```
[the remark this answers is not in the document]
Comment (micoli) [reply data/sylar.yaml:17]: it was not introduced here, it was fixed here :)

**Comment by micoli** [comment]
Asked changes are applied. Could you please validate them. Thanks for your time

**Comment by sereinity** [comment]
> Asked changes are applied. Could you please validate them. Thanks for your time

I think I will take a look tomorrow or Thursday.

**Review by sereinity — APPROVED** [review]

**Pull request merged**